Improve accessibility and consistency of the 'Last modified' Revisions button.

afercia opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->
Fixes https://github.com/WordPress/gutenberg/issues/66526
Alternative to https://github.com/WordPress/gutenberg/pull/66535

## What?
<!-- In a few words, what is the PR actually doing? -->
The 'Last modified' button in the Site editor navigation panel > Styles is not accessible and arguably usable as the visibel text mismatches the accessible name and the button's action isn't clear.

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->
A complete mismatch between an interactive control visible text and its accessible name is a WCAG violation.
On top of that, buttons must clearly communicate the udnerlying action for all users.
For more details, see the issue https://github.com/WordPress/gutenberg/issues/66526

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->
Separates the 'last modified' information from the button.
Makes clearer the button is a navigation button and that it brings to 'Revisions'.

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a post or page. -->
<!-- 2. Insert a heading block. -->
<!-- 3. etc. -->

### Old instructions
- Make sure you have some Styles revisions saved. Make a global styles change and save it if necessary.
- Go to the Site editor > Styles 
- Observe the navigation button at the bottom of the navigation panel.
- Observe the 'Last modified' information is now separate from the navigation button.
- Observe the navigation button has the revisions icon and visible text 'Revisions'.

### New instructions
- Make sure you have no Styles revisions yet, either by deleting them from the database (I think the post type is `wp_global_styles`?) or by switching to a theme where you haven't saved Styles changes yet.
- Go to the Site editor > Styles 
- Observe at the bottom of the black navigation panel there's no 'revisions' button.
- Make a global styles change and save it.
- Go back to the 'view' mode of the editor by clicking the 'Open navigation' in the top left of the screen.
- Observe at the bottom of the black navigation panel there's now the Revisions button rendered, with the count of saved revisions: 1.
- Make more styles changes, save them, and check the count is always correct.
- Optionally: switch to another theme and check the revisions count is correct for that theme.

### Testing Instructions for Keyboard
<!-- How can you test the changes by using the keyboard only? Please note, this is required for PRs that change the user interface (UI). This ensures the PR can be tested for any possible accessibility regressions. -->

## Screenshots or screencast <!-- if applicable -->

| Before | Outdated | After |
| ------------- | ------------- | ------------- |
| ![before 01](https://github.com/user-attachments/assets/1ed84919-66a8-48f4-b18c-d93de1bb8ee2)  | ![after](https://github.com/user-attachments/assets/287be2f6-a54d-491c-9599-f354f0f8345a)  | ![Screenshot 2024-11-20 at 14 52 58](https://github.com/user-attachments/assets/e0be9ba7-48be-4d67-b3c1-53c76a913a65)

afercia committed:
Improve accessibility and consistency of the Revisions button.

afercia commented:
A note on teh following components used for this UI:

- SidebarNavigationScreenDetailsPanelRow
- SidebarNavigationScreenDetailsPanelLabel
- SidebarNavigationScreenDetailsPanelValue

They are all wrapper to other components like HStack and Text. It appears to me they are a bit overkill and not really necessary. It looks like they're only used here and the more generic component could be used instead.

Also, [SidebarNavigationScreenDetailsPanelLabel](https://github.com/WordPress/gutenberg/blob/7714bce8d9356707a020b5d7df84797832c423f8/packages/edit-site/src/components/sidebar-navigation-screen-details-panel/sidebar-navigation-screen-details-panel-label.js) and [SidebarNavigationScreenDetailsPanelValue](https://github.com/WordPress/gutenberg/blob/7714bce8d9356707a020b5d7df84797832c423f8/packages/edit-site/src/components/sidebar-navigation-screen-details-panel/sidebar-navigation-screen-details-panel-value.js) are basically duplicate.

The code could be greatly simplified by entirely removing these components.

afercia commented:
Rebased.
@WordPress/gutenberg-core I'd appreciate a review, when you have a chance.``

ramonjd commented on packages/edit-site/src/components/sidebar-navigation-screen-details-footer/index.js:
```diff
@@ -44,33 +43,32 @@ export default function SidebarNavigationScreenDetailsFooter( {
 							sprintf(
 								/* translators: %s: is the relative time when the post was last modified. */
-								__( '<time>%s</time>' ),
+								__( 'Last modified: <time>%s</time>' ),
```
Do you think it makes sense to use the [simpler format used by posts and templates](https://github.com/WordPress/gutenberg/blob/564f0699e0924b7114af55d51f1909de07a2e3c9/packages/editor/src/components/post-last-edited-panel/index.js#L20-L20)?


<img width="245" alt="Screenshot 2024-11-20 at 10 04 22 am" src="https://github.com/user-attachments/assets/687b6216-871f-4849-8a3f-06b3d30bc323">

```js
sprintf(
	// translators: %s: Human-readable time difference, e.g. "2 days ago".
	__( 'Last edited %s.' ),
	humanTimeDiff( modified )
);
```

ramonjd replied:
Actually, on second thought. How valuable is this information?

If it's still accessible, I think just having the revisions button with a count might appear less cluttered:

```js
// translators: %s: number of revisions.
sprintf( __( '(%s) Revisions' ), revisionsCount )
```

![Screenshot 2024-11-20 at 10 19 44 am](https://github.com/user-attachments/assets/81dff5c2-5fec-4889-85d0-30c0771471c2)

What do you reckon?

ramonjd reviewed (commented):
Thanks for the PR. I don't have much to add, other than I agree the component could be simplified. 

Keen to learn what @jameskoster thinks.

afercia commented:
I wouldn't be opposed to remove the 'Last modified' info if there's agreement on that.
After all, after clicking on 'Revisions', users will see the list of revisions and can see the date and time of the last edit.

afercia commented:
>  just having the revisions button with a count might appear less cluttered

I'm going to try that soon, thanks.

t-hamano commented on packages/edit-site/src/components/sidebar-navigation-screen-details-footer/index.js:
```diff
@@ -17,7 +17,6 @@ import { backup } from '@wordpress/icons';
 import {
 	SidebarNavigationScreenDetailsPanelRow,
 	SidebarNavigationScreenDetailsPanelLabel,
-	SidebarNavigationScreenDetailsPanelValue,
```
Since the `SidebarNavigationScreenDetailsPanelValue` component isn't being used anywhere anymore, we should be able to delete the component itself.

[packages/edit-site/src/components/sidebar-navigation-screen-details-panel/sidebar-navigation-screen-details-panel-value.js](https://github.com/WordPress/gutenberg/blob/fix/styles-navigation-revisions-button/packages/edit-site/src/components/sidebar-navigation-screen-details-panel/sidebar-navigation-screen-details-panel-value.js)

t-hamano commented on packages/edit-site/src/components/sidebar-navigation-screen-details-panel/style.scss:
```diff
@@ -17,10 +17,9 @@
 }
 
+.edit-site-sidebar-navigation-details-screen-panel__label time,
 .edit-site-sidebar-navigation-details-screen-panel__value.edit-site-sidebar-navigation-details-screen-panel__value {
```
We can remove the `.edit-site-sidebar-navigation-details-screen-panel__value` CSS class.

t-hamano commented on packages/edit-site/src/components/sidebar-navigation-screen/style.scss:
```diff
@@ -127,9 +127,14 @@
 	margin: $grid-unit-20 0 0;
 	border-top: 1px solid $gray-800;
+
+	.components-item-group {
```
```suggestion
	.edit-site-sidebar-navigation-screen-details-footer {
```

Let's avoid using `.components`-prefixed CSS classes whenever possible.

t-hamano reviewed (commented):
Thanks for the PR.

This PR will probably be updated, but I'll comment on what I've noticed so far.

afercia committed:
Simplify revisions button and add count.

afercia commented:
In the last commit I'm trying to simplify a lot, remove unused components and CSS, and add the revisions count. Screenshot:

![Screenshot 2024-11-20 at 14 52 58](https://github.com/user-attachments/assets/3e5cfce1-8c9c-4a7b-9cc4-ae5b922fcc9e)

I'll update the testing instructions.

afercia committed:
Remove no longer used stylesheet import.

ramonjd commented:
> appears to be 0 even when there are saved revisions. It seems to me that _links_ is only populated for an unsaved revision.

Thanks for flagging that. It does seems odd. I'll look into it 🙇🏻

afercia commented:
Thank you @ramonjd. I will have a look at https://github.com/WordPress/gutenberg/pull/67180 and wait for it to be merged, then adapt this PR.

afercia committed:
Simplify revisions count retrieval by using useGlobalStylesRevisions.

afercia commented:
Last commit simplifies the revisions count retrieval following the approach from https://github.com/WordPress/gutenberg/pull/67180

afercia committed:
Adjust after 67180.

afercia committed:
Remove duplicate import after rebase.

afercia commented:
I adjusted this PR again following the new approach to retrieve the revisions count from https://github.com/WordPress/gutenberg/pull/67180 and rebased. @ramonjd I'd appreciate a review, when you have a chance 🙏🏻

ramonjd reviewed (approved):
This is a good change in my opinion. The last modified information was superfluous, and the reduced content appears much neater.

I tested in Firefox, Safari and Chrome. No regressions in the link to the revision panel.

Here's what I'm seeing:


| No revisions | One revision | > 1 revisions |
|--------|--------|--------|
| <img width="300" alt="Screenshot 2024-11-26 at 10 03 04 am" src="https://github.com/user-attachments/assets/1f0267e6-acef-467d-a644-382f66fcc2b7"> | <img width="300" alt="Screenshot 2024-11-26 at 10 03 25 am" src="https://github.com/user-attachments/assets/d8cb61d4-d6d6-4202-8e7e-a95238812b21"> | <img width="300" alt="Screenshot 2024-11-26 at 10 03 45 am" src="https://github.com/user-attachments/assets/656d4691-0e0b-49ce-a251-43128d39852b"> | 


Here's what's in trunk right now:

<img width="300" alt="Screenshot 2024-11-26 at 10 13 03 am" src="https://github.com/user-attachments/assets/cc42d4ce-a3e5-49cc-ab06-21ca953c41d6">

afercia commented:
Thanks @ramonjd for your review.

> The last modified information was superfluous, and the reduced content appears much neater.

It's worth reminding that besides the the visual changes, this is mainly an accessibility fix to avoid a mismatch between visible text and accessible name. For example:

<img width="293" alt="last modified" src="https://github.com/user-attachments/assets/7234a2b9-bb12-4297-b030-27e4df2d2bca">

- Visible text: "Last modified 9 minutes ago"
- Actual accessible name: "Revisions"

This kind of mismatch is not OK. I would appreciate any new design to take into consideration that the visual labeling of interactive controls is not a place to communicate values or states. The visual labeling must communicate _what_ the control does. Attempting to remediate with an aria-label (in this case it was 'Revisions') doesn't make things better. Instead, it introduces a barrier for users and a WCAG violation.

afercia commented:
Thanks @jameskoster, looks like changing [this selector after core review](https://github.com/WordPress/gutenberg/pull/66606#discussion_r1850146205) is the culprit of the alignment regression. Will submit a quick PR to fix it.
